app/src/main/java/com/quarkworks/apartmentgroceries/grocery/GroceryCell.java

```diff
@@ -44,13 +58,37 @@ private void initialize() {
         LayoutInflater.from(getContext()).inflate(R.layout.grocery_cell, this);
         nameTextView = (TextView) findViewById(R.id.grocery_cell_grocery_name_id);
         createdByTextView = (TextView) findViewById(R.id.grocery_cell_created_by_id);
+        createdAtTextView = (TextView) findViewById(R.id.grocery_cell_created_at_id);
         purchasedByTextView = (TextView) findViewById(R.id.grocery_cell_purchased_by_id);
+        statusTextView = (TextView) findViewById(R.id.grocery_cell_grocery_status_id);
+        photoImageView = (ImageView) findViewById(R.id.grocery_cell_created_by_image_view_id);
     }
 
-    public void setViewData(final RGroceryItem groceryItem, final int position){
+    public void setViewData(final RGroceryItem groceryItem, final int position) {
         nameTextView.setText(groceryItem.getName());
-        createdByTextView.setText(groceryItem.getCreatedBy());
-        purchasedByTextView.setText(groceryItem.getPurchasedBy());
+        createdAtTextView.setText(Utilities.getReadableDate(groceryItem.getCreatedAt(), null, null));
+
+        String purchasedBy = groceryItem.getPurchasedBy();
+        if (TextUtils.isEmpty(purchasedBy)) {
+            statusTextView.setText(getResources().getString(R.string.grocery_cell_item_status_open));
+        } else {
+            purchasedByTextView.setText(purchasedBy);
+            statusTextView.setText(getResources().getString(R.string.grocery_cell_item_status_closed));
+        }
+        statusTextView.setTextColor(getResources().getColor(R.color.colorPrimary));
+
+        photoImageView.setImageResource(R.drawable.ic_launcher);
+
+        final RUser rUser = DataStore.getInstance().getRealm().where(RUser.class)
+                .equalTo(SyncUser.JsonKeys.USER_ID, groceryItem.getCreatedBy()).findFirst();
+        createdByTextView.setText(rUser.getUsername());
+
```

Comment:
Since we don't necessarily have this user in the database yet, we need to do a null check here on rUser before setting the view data or it will crash.